ownCloud/SDK Extensions/OCItem+Extension.swift

```diff
@@ -213,7 +213,16 @@ extension OCItem {
 		return nil
 	}
 
-	func fileSuffix() -> String {
-		return (self.name as NSString).pathExtension
+	func fileExtension() -> String {
+		var fileExtension = (self.name as NSString).pathExtension
+		if fileExtension != "" {
+			fileExtension = ".\(fileExtension)"
```

Comment:
The property `name` in `OCItem` has the extension in it so for example it would return `foobar.pfd`. If you want to make a duplicate file, you need to `copy` the file and give it a new name. Does not work simply by adding "copy" at the end because with no extension the file loses its mime-type.

I would prefer that this method returns the dot because if not, we need to add it later with some code like the following:
```Swift
if item.type != .collection {
	let itemName = item.nameWithoutExtension()
	var fileExtension = item.fileExtension()

	if fileExtension != "" {
		fileExtension = ".\(fileExtension)"
	}

	name = "\(itemName) copy\(fileExtension)"
}
```

Reply:
@pablocarmu I understood that but I proposed not to add the dot for the sake of the clean API since dot is not actually a part of the file extension it is just a separator. Ideally, you should be able to understand what the API function is returning without looking up the documentation and I personally would not expect fileExtension() to return extension plus something else.